WordPress/gutenberg
# [Mobile] Update toolbar icons and colors

**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5940

## What?
This PR updates the current colors for the toolbar icons, the toolbar itself, and the safe area background.

## Why?
It's part of the improvements for the mobile editor's UI.

## How?
- It uses the current colors available in the `_colors.native.scss` file for the updates
- By adding a new color variable `$app-safe-area-background` for the SafeArea
- Updates the Button component with new styles for the active states. It **does not** refactor the component to extract inline styles but it adds the styles file with the new additions.
- Adds a drop shadow to the "Hide keyboard" button, for Android a solid color was needed or it wouldn't show up.
- Updates the text color formatting to show two different icons depending on if the selected content has a color set or not.
- Restores the BlockSettings button logic in https://github.com/WordPress/gutenberg/pull/52336/commits/4ed6b7a5d47eb8e406b5692601d87be40c0d4ce0 which was removed in https://github.com/WordPress/gutenberg/pull/52289 but it was needed at the end to prevent showing the BlockSettings button when the selected block doesn't have any settings.

## Testing Instructions

- Open the app
- When there's no block selected or the title is selected
- **Expect** to see the add quick blocks icons with the new styles
- Add a Paragraph block
- Type some text
- Add some text formatting
- **Expect** the formatting icons show the new styles for both light/dark modes
- Add a block that is not a rich text one e.g. Columns, Cover
- **Expect** to see the block's icons in the toolbar to show the new styles

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 

iPhone Quick Media Icons Light mode | iPhone Quick Media Icons Dark mode
-|-
<kbd><img src="https://github.com/WordPress/gutenberg/assets/4885740/d4827ead-822d-43a3-8d1d-eb886d234c72" width="250" /></kbd> | <kbd><img src="https://github.com/WordPress/gutenberg/assets/4885740/4f8546fc-2665-483f-a6ff-b90c50a7e186" width="250" /></kbd>

iPhone Format Icons Light mode | iPhone Format Icons Dark mode
-|-
<kbd><img src="https://github.com/WordPress/gutenberg/assets/4885740/01d65396-2fca-4662-beb6-981c50852119" width="250" /></kbd> | <kbd><img src="https://github.com/WordPress/gutenberg/assets/4885740/883c83f8-6503-42ef-ab20-612edee9d1f9" width="250" /></kbd>

iPhone Block Icons Light mode | iPhone Block Icons Dark mode
-|-
<kbd><img src="https://github.com/WordPress/gutenberg/assets/4885740/b95c272b-d963-420d-971b-03ac2f1a3a27" width="250" /></kbd> | <kbd><img src="https://github.com/WordPress/gutenberg/assets/4885740/972bc193-6deb-40c9-a6b8-0a53288b768f" width="250" /></kbd>

Android Quick Media Icons Light mode | Android Quick Media Icons Dark mode
-|-
<kbd><img src="https://github.com/WordPress/gutenberg/assets/4885740/2f13c947-8326-4f74-a707-e8232c769e25" width="250" /></kbd> | <kbd><img src="https://github.com/WordPress/gutenberg/assets/4885740/5c4cd349-a847-4c5a-8e21-6e6c547063dd" width="250" /></kbd>

Android Format Icons Light mode | Android Format Icons Dark mode
-|-
<kbd><img src="https://github.com/WordPress/gutenberg/assets/4885740/22ee9bf1-d531-41a7-b93d-66e43ff4e363" width="250" /></kbd> | <kbd><img src="https://github.com/WordPress/gutenberg/assets/4885740/4ef26a99-e582-4c13-9f93-1a9f85ccabd9" width="250" /></kbd>

Android Block Icons Light mode | Android Block Icons Dark mode
-|-
<kbd><img src="https://github.com/WordPress/gutenberg/assets/4885740/eb17ad33-0c11-4f39-99e8-8e0cfa310f6e" width="250" /></kbd> | <kbd><img src="https://github.com/WordPress/gutenberg/assets/4885740/a55d5fdb-75c0-4afc-af18-6b50bac8630c" width="250" /></kbd>
